photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/DataProcessingUtils.scala

```diff
@@ -36,94 +36,71 @@ object DataProcessingUtils {
       featureShardIdToFeatureMapLoader: Map[String, IndexMapLoader]): Map[String, Int] = {
 
     featureShardIdToFeatureMapLoader.map { case (shardId, featureMapLoader) =>
-      (shardId, featureMapLoader.indexMapForDriver.featureDimension)
+      (shardId, featureMapLoader.indexMapForDriver().featureDimension)
     }
   }
 
-  //TODO: Change the scope to protected[avro] after Avro related classes/functIons are decoupled from the rest of code
+  /**
+   * Parse a [[RDD]] of type [[GameDatum]] from a [[RDD]] of type [[GenericRecord]]
+   * @param records a [[RDD]] of type [[GenericRecord]]
+   * @param featureShardIdToFeatureSectionKeysMap a map from feature shard id (defined by the user) to feature
+   *                                              section keys (define in input data's Avro schema)
+   * @param featureShardIdToFeatureMapLoader a map from feature shard id (defined by the user) to the feature map loader
```

Comment:
This doc doesn't offer any more information than the parameter name. What is a FeatureMapLoader?